src/search/index.test.ts

```diff
@@ -0,0 +1,13 @@
+import * as search from ".";
+
+test("empty string", () => {
+  let s: string = "";
+
+  expect(search.Search(s)).toHaveLength(0);
+});
```

Comment:
Note the first basic test cases. These will be extended further. One test case at a time.